test/extensions/transport_sockets/alts/tsi_socket_test.cc

```diff
@@ -382,6 +383,152 @@ TEST_F(TsiSocketTest, HandshakeWithInternalError) {
   raw_handshaker->vtable = vtable;
 }
 
+TEST_F(TsiSocketTest, DoReadEndOfStream) {
+  initialize(nullptr, nullptr);
+
+  client_.write_buffer_.add(ClientToServerData);
+
+  doHandshakeAndExpectSuccess();
+
+  EXPECT_CALL(*client_.raw_socket_, doWrite(_, false));
+  expectIoResult({Network::PostIoAction::KeepOpen, 21UL, false},
+                 client_.tsi_socket_->doWrite(client_.write_buffer_, false));
+
+  EXPECT_CALL(*server_.raw_socket_, doRead(_)).WillOnce(Invoke([&](Buffer::Instance& buffer) {
+    Network::IoResult result = {Network::PostIoAction::KeepOpen, client_to_server_.length(), true};
+    buffer.move(client_to_server_);
+    return result;
+  }));
+  EXPECT_CALL(server_.callbacks_, shouldDrainReadBuffer());
+  expectIoResult({Network::PostIoAction::KeepOpen, 21UL, true},
+                 server_.tsi_socket_->doRead(server_.read_buffer_));
+
+  EXPECT_EQ(ClientToServerData, server_.read_buffer_.toString());
+}
+
+TEST_F(TsiSocketTest, DoReadNoData) {
+  initialize(nullptr, nullptr);
+
+  client_.write_buffer_.add(ClientToServerData);
+
+  doHandshakeAndExpectSuccess();
+
+  EXPECT_CALL(*client_.raw_socket_, doWrite(_, false));
+  expectIoResult({Network::PostIoAction::KeepOpen, 21UL, false},
+                 client_.tsi_socket_->doWrite(client_.write_buffer_, false));
+
+  EXPECT_CALL(*server_.raw_socket_, doRead(_))
+      .WillOnce(Invoke([&](Buffer::Instance& buffer) {
+        Network::IoResult result = {Network::PostIoAction::KeepOpen, client_to_server_.length(),
+                                    false};
+        buffer.move(client_to_server_);
+        return result;
+      }))
+      .WillOnce(Invoke([&](Buffer::Instance& buffer) {
+        Network::IoResult result = {Network::PostIoAction::KeepOpen, 0UL, false};
+        buffer.move(client_to_server_);
+        return result;
+      }));
+  EXPECT_CALL(server_.callbacks_, shouldDrainReadBuffer());
+  expectIoResult({Network::PostIoAction::KeepOpen, 21UL, false},
+                 server_.tsi_socket_->doRead(server_.read_buffer_));
+
+  EXPECT_EQ(ClientToServerData, server_.read_buffer_.toString());
+}
+
+TEST_F(TsiSocketTest, DoReadError) {
+  initialize(nullptr, nullptr);
```

Comment:
I see that server_.raw_socket_ is a NiceMock, should we consider removing NiceMock in order to get the StrictMock default in these tests?

https://github.com/envoyproxy/envoy/blob/a7f46e02a0857926aace227fce2056164b5a6f10/test/extensions/transport_sockets/alts/tsi_socket_test.cc#L47

Reply:
Done.